homeassistant/components/control4/config_flow.py

```diff
@@ -0,0 +1,163 @@
+"""Config flow for Control4 integration."""
+from asyncio import TimeoutError as asyncioTimeoutError
+import logging
+
+from aiohttp.client_exceptions import ClientError
+from pyControl4.account import C4Account
+from pyControl4.director import C4Director
+from pyControl4.error_handling import NotFound, Unauthorized
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import CONF_CONTROLLER_UNIQUE_ID, DEFAULT_SCAN_INTERVAL, MIN_SCAN_INTERVAL
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class Control4Validator:
+    """Validates that config details can be used to authenticate and communicate with Control4."""
+
+    def __init__(self, host, username, password):
+        """Initialize."""
+        self.host = host
+        self.username = username
+        self.password = password
+        self.account = None
+        self.controller_unique_id = None
+        self.director_bearer_token = None
+        self.director = None
```

Comment:
The `account` and `director` variables don't need to be instance attributes as they are local to the respective methods.